public/app/features/browse-dashboards/components/DashboardsTree.tsx

```diff
@@ -0,0 +1,165 @@
+import { css, cx } from '@emotion/css';
+import React, { useMemo } from 'react';
+import { CellProps, Column, TableInstance, useTable } from 'react-table';
+import { FixedSizeList as List } from 'react-window';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Checkbox, useStyles2 } from '@grafana/ui';
+
+import { DashboardsTreeItem, INDENT_AMOUNT_CSS_VAR } from '../types';
+
+import { NameCell } from './NameCell';
+import { TypeCell } from './TypeCell';
+
+interface DashboardsTreeProps {
+  items: DashboardsTreeItem[];
+  width: number;
+  height: number;
+  onFolderClick: (uid: string, newOpenState: boolean) => void;
+}
+
+type DashboardsTreeColumn = Column<DashboardsTreeItem>;
+
+const HEADER_HEIGHT = 35;
+const ROW_HEIGHT = 35;
+
+export function DashboardsTree({ items, width, height, onFolderClick }: DashboardsTreeProps) {
+  const styles = useStyles2(getStyles);
+
+  const tableColumns = useMemo(() => {
+    const checkboxColumn: DashboardsTreeColumn = {
+      id: 'checkbox',
+      Header: () => <Checkbox value={false} />,
+      Cell: () => <Checkbox value={false} />,
```

Comment:
Currently the checkboxes don't do anything. Should a handler be passed here?

Reply:
> Item selection is not implemented. Will probably happen in seperate PR.